source/mir/random/engine/mersenne_twister.d

```diff
@@ -105,7 +106,7 @@ struct MersenneTwisterEngine(UIntType, size_t w, size_t n, size_t m, size_t r,
         // them separately in sequence, the variables
         // are kept 'hot' in CPU registers, allowing
         // for significantly faster performance.
-        sizediff_t index = cast(size_t)this.index;
+        sizediff_t index = this.index;
```

Comment:
I'm not sure I follow your argument here.  I'm only concerned about the _naming_ of the local `index` variable being different from the _naming_ of the struct field it takes the value of.